### PR TITLE
feat(cfb): link sub- & rowforms in question list

### PR DESCRIPTION
**Pull request by luytena**
For table and form questions add direct link to subform
or rowform in question list. Show correct path and
question list corresponding to sub- / rowform.

Resolves: #1416

image

### CODE REVIEW

File: addon/components/cfb-form-editor/question-list/item.hbs [hunk 1]
```diff
@@ -50,11 +50,27 @@
...
+       uk-margin-small-right uk-text-small uk-text-muted uk-text-truncate"
+  >
+    {{@question.label}}
+  </span>
+  {{#if (or isForm isTable)}}
+    <span uk-icon="link" class="uk-flex-none"></span>
+    <a data-test-link-subform
+      href="#"
+      class="uk-width-expand uk-margin-small-right uk-text-small uk-text-truncate"
+      {{on "click" (fn (optional @on-retrieve-form) (or @question.subForm @question.rowForm))}}
```
Comment (anehx) [inline addon/components/cfb-form-editor/question-list/item.hbs:68]: The action name is not really intuitive. How about `on-click-form`?

**Review by anehx — CHANGES_REQUESTED** [review]
Minor nitpick, otherwise LGTM :fire:

File: addon/components/cfb-form-editor/question-list/item.hbs (hunk 1 above)
Comment (luytena) [reply addon/components/cfb-form-editor/question-list/item.hbs]: Sure that works, I just updated it in all locations :+1:

**Comment by anehx** [comment]
@luytena needs a rebase for the CI to work

**Comment by luytena** [comment]
> @luytena needs a rebase for the CI to work

done, works now

**Review by anehx — APPROVED** [review]

**Comment by luytena** [comment]
Tests failing because the question list is reset to reorder mode when the form is changed (at the moment also when a new question form is opened). It expects the question list to remain in the same mode after for example adding a question.

What is the expected behavior when following a link to another form while in the add / remove mode? Should the question list remain in the add / remove mode or reset to reorder mode? Originally I reset it to reorder mode to ensure that users are aware of the question list in the new form, and don't edit the question list assuming they are still editing the previous form.

**Comment by anehx** [comment]
@luytena I think we should just display the link in reorder mode. When adding or deleting a question one normally doesn't want to check out the linked form ^^

**Comment by luytena** [comment]
> @luytena I think we should just display the link in reorder mode. When adding or deleting a question one normally doesn't want to check out the linked form ^^

@anehx Makes sense, I've updated it accordingly :+1:

File: addon/components/cfb-form-editor/question-list/item.hbs [hunk 2]
```diff
@@ -50,11 +50,33 @@
...
+{{#let
+  (and 
+    (eq @mode "reorder")
+    (or
+      (has-question-type @question "form")
+      (has-question-type @question "table")
+    )
+  )
+  as |showFormLink|
+}}
```
Comment (anehx) [inline addon/components/cfb-form-editor/question-list/item.hbs:62]: Let's put this in the JS file. This makes the template hard to read..

Comment (luytena) [reply addon/components/cfb-form-editor/question-list/item.hbs]: Fixed

**Review by anehx — APPROVED** [review]

**Pull request merged**